public/app/features/dashboard-scene/saving/provisioned/SaveProvisionedDashboard.tsx

```diff
@@ -72,23 +75,36 @@ export function SaveProvisionedDashboard({ drawer, changeInfo, dashboard }: Prop
     const appEvents = getAppEvents();
     if (request.isSuccess) {
       dashboard.setState({ isDirty: false });
-      if (isNew) {
-        dashboard.setManager(ManagerKind.Repo, defaultValues.repo);
-      }
       if (workflow === 'branch' && ref !== '' && path !== '') {
         // Redirect to the provisioning preview pages
         navigate(`${PROVISIONING_URL}/${defaultValues.repo}/dashboard/preview/${path}?ref=${ref}`);
-      } else {
-        appEvents.publish({
-          type: AppEvents.alertSuccess.name,
-          payload: ['Dashboard changes saved'],
-        });
-        dashboard.closeModal();
-        locationService.partial({
-          viewPanel: null,
-          editPanel: null,
-        });
+        return;
+      }
+
+      appEvents.publish({
+        type: AppEvents.alertSuccess.name,
+        payload: ['Dashboard changes saved'],
+      });
+
+      // Load the new URL
+      const upsert = request.data.resource.upsert as Resource<Dashboard>;
+      if (isNew && upsert?.metadata?.name) {
+        navigate(
+          getDashboardUrl({
+            uid: upsert.metadata.name,
+            slug: kbn.slugifyForUrl(upsert.spec.title ?? ''),
+            currentQueryParams: '', // ???
+          })
+        );
+        return;
```

Comment:
```suggestion
      if (isNew && upsert?.metadata?.name) {
        const url = locationUtil.assureBaseUrl(
          getDashboardUrl({
            uid: upsert.metadata.name,
            slug: kbn.slugifyForUrl(upsert.spec.title ?? ''),
            currentQueryParams: '',
          })
        );

        navigate(url);
        return;
```
Can't see a nicer way to do this, but I think we need to use `locationUtil.assureBaseUrl` in case Grafana is served from a subpath.